### PR TITLE
add autofocus on change pin screen and remove dots when back from cha…

### PR DESCRIPTION
**Pull request by mateuszpmroz**
…nging the pin

### CODE REVIEW

File: src/components/PinInput.tsx [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
 export class PinInput extends PureComponent<Props> {
   codeFieldRef = React.createRef<any>();
+  onFocus: Function = noop;
 
   componentDidMount() {
     // workaround for autoFocus prop issues
-    setTimeout(() => {
-      this.focus();
-    }, 500);
+    this.onFocus = this.props.navigation.addListener('focus', () => {
```
Comment (KedziaPawel) [inline src/components/PinInput.tsx:26]: what about removing this listener?

Comment (mateuszpmroz) [reply src/components/PinInput.tsx:26]: > what about removing this listener?

It's removed in `componentWillUnmount` function.
https://reactnavigation.org/docs/navigation-events/

Comment (KedziaPawel) [reply src/components/PinInput.tsx:26]: ok I see, could you name it unsubscribeOnFocus?

Comment (mateuszpmroz) [reply src/components/PinInput.tsx:26]: > ok I see, could you name it unsubscribeOnFocus?

maybe just `focusListener` to be sure that the variable is a listener?

Comment (KedziaPawel) [reply src/components/PinInput.tsx:26]: I would like to see that this is a function to unsubscribe, so maybe `unsubscribeFocusListener`?

Comment (mateuszpmroz) [reply src/components/PinInput.tsx]: ok, changed

**Pull request merged**